packages/core/src/concurrent/pool.ts

```diff
@@ -12,7 +12,7 @@ const MAX_CONCURRENT_INIT = 2;
 /**
  * Represents a Checker that is also a Resource (with an init and dispose)
  */
-export type CheckerResource = Checker & Resource;
+export type CheckerResource = ActiveChecker & Checker & Resource;
```

Comment:
See the previous remark. I don't think a `CheckerResource` should be ` Checker` anymore.